src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1933,6 +1934,20 @@ private void OnFileChanged(FileSystemEventArgs e)
                 ScheduleRestartAsync(shutdown).ContinueWith(t => TraceWriter.Error($"Error restarting host (full shutdown: {shutdown})", t.Exception),
                     TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnFaulted);
             }
+            else
```

Comment:
Instead of adding this branch, can we make sure the logic is improved to set the `changeDescription` instead? That should be a simpler change and will avoid the code duplication